src/instrumentsscene/view/layoutpaneltreemodel.cpp

```diff
@@ -553,6 +565,17 @@ bool LayoutPanelTreeModel::moveRows(const QModelIndex& sourceParent, int sourceR
     sourceParentItem->moveChildren(sourceFirstRow, count, destinationParentItem, destinationRow, !m_dragInProgress);
     endMoveRows();
 
+    int lastRow = m_rootItem->childCount() - 1;
+    AbstractLayoutPanelTreeItem* lastItem = m_rootItem->childAtRow(m_rootItem->childCount() - 1);
+    if (lastItem->type() == LayoutPanelItemType::SYSTEM_OBJECTS_LAYER) {
```

Comment:
Could you please explain when this might happen? updateSystemObjectLayers() already removes unnecessary items, doesn't it?